src/main.rs

```diff
@@ -20,6 +20,15 @@ fn positive_integer(val: String) -> Result<(), String> {
     u32::from_str(&val).map(|_| ()).map_err(|e| format!("{e}"))
 }
 
+fn label_list(val: String) -> Result<(), String> {
+    val.chars()
+        .all(|c| {
+            c == ',' || (c.is_ascii() && c.is_alphanumeric()) || c == '-' || c == '_' || c == '.'
+        })
+        .then_some(())
+        .ok_or("Expected list of Metamaths labels".to_string())
```

Comment:
```suggestion
        .ok_or("Expected list of labels".to_string())
```
I don't think we need to specify this, we don't usually qualify the term elsewhere. We could clarify the grammar in the error message if you think it helps though.

Reply:
This is also a good place to use glob patterns, if we have them (I think david had an implementation in another PR?)

Reply:
We did not keep the glob patterns in the final solution. I think eventually the best is to first split the project into library/command line tool, and then only include the glob library of choice into the latter. So yes, let's keep that for later.